File: 프로그래머스/lv1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.js
function solution(s) {
    var answer = s;
    const numberList = ['zero','one','two','three','four','five','six','seven','eight','nine'];
    numberList.forEach((el,index)=>{
        answer =answer.replaceAll(el,index);
    })
    
    return Number(answer);
}

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java
import java.util.*;

class Solution {
    public int solution(String s) {
        String[] arr = {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
        Map<String, String> map = new HashMap<>();
        for(int i=0 ;i<10; i++){
            map.put(arr[i], String.valueOf(i));
        }
        
        for(String target : arr){
            s = String.join(map.get(target), s.split(target,-1));
        }
        
        return Integer.valueOf(s);
    }
}